src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -90,8 +90,13 @@ public static void Run(this IWebHost host)
                     done.Wait();
                 };
 
+#if NETCOREAPP2_0
                 var assemblyLoadContext = AssemblyLoadContext.GetLoadContext(typeof(WebHostExtensions).GetTypeInfo().Assembly);
                 assemblyLoadContext.Unloading += context => shutdown();
```

Comment:
Looks like it works, at least the functional test passes. This would simplify the PR a bit.